server/parser.go

```diff
@@ -26,6 +30,7 @@ type parseState struct {
 	pa      pubArg
 	argBuf  []byte
 	msgBuf  []byte
+	header  http.Header // access via getHeader
```

Comment:
Would have put that under `pubArg` along with `hdr`, but maybe it needs to be there, not sure.